gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProc.java

```diff
@@ -63,13 +64,16 @@ public DagProc(DagTask dagTask) {
     this.dagNodeId = this.dagTask.getDagAction().getDagNodeId();
   }
 
-  public final void process(DagManagementStateStore dagManagementStateStore) throws IOException {
-    T state = initialize(dagManagementStateStore);   // todo - retry
-    act(dagManagementStateStore, state);   // todo - retry
+  public final void process(DagManagementStateStore dagManagementStateStore,
+      DagProcessingEngineMetrics dagProcEngineMetrics) throws IOException {
+    T state = initialize(dagManagementStateStore, dagProcEngineMetrics);   // todo - retry
+    act(dagManagementStateStore, state, dagProcEngineMetrics);   // todo - retry
```

Comment:
I wrapped the calls for `initialize` since it's a relatively simple function, but `act` would be a little more complicated because there are multiple points of failure. I would have to change the API to return a boolean perhaps indicating whether or not `act` was successful because there are failure cases that do not result in an exception being thrown.